gatewayconfig/gatewayconfig_app.py

```diff
@@ -38,15 +40,18 @@ def __init__(self, sentry_dsn, balena_app_name, balena_device_uuid, variant, eth
         eth0_mac_address = read_eth0_mac_address(eth0_mac_address_filepath)
         wlan0_mac_address = read_wlan0_mac_address(wlan0_mac_address_filepath)
         logger.debug("Read eth0 mac address %s and wlan0 %s" % (eth0_mac_address, wlan0_mac_address))
-        pub_key, onboarding_key, animal_name = read_miner_keys(miner_keys_filepath)
-        logger.debug("Read onboarding pub_key: %s + animal_name: %s" % (pub_key, animal_name))
 
         self.bluetooth_services_processor = BluetoothServicesProcessor(eth0_mac_address, wlan0_mac_address, onboarding_key, pub_key, firmware_version, ethernet_is_online_filepath, self.shared_state)
         self.led_processor = LEDProcessor(self.status_led, self.shared_state)
         self.diagnostics_processor = DiagnosticsProcessor(diagnostics_json_url, self.shared_state)
         self.wifi_processor = WifiProcessor(self.shared_state)
         self.bluetooth_advertisement_processor = BluetoothAdvertisementProcessor(eth0_mac_address, self.shared_state, self.variant_details)
 
+        pub_key = self.diagnostics_processor['PK']
```

Comment:
This won't work. `diagnostics_processor` is a class, not JSON. To keep things moving, I'd recommend not using diagnostics_processor and instead some minimal code duplication.

```
diagnostics_response = requests.get(self.diagnostics_json_url)
diagnostics_json = response.json()
pub_key = diagnostics_json['PK']
onboarding_key = diagnostics_json['OK']
animal_name = diagnostics_json['AN']
```